src/server/mod.rs

```diff
@@ -1324,30 +1334,39 @@ impl Server {
         false
     }
 
-    #[allow(unused_must_use)]
-    pub fn on_left_click(&self) {
-        let mut entities = self.entities.write();
-        let mut mouse_buttons = entities
-            .world
-            .entity_mut(self.player.read().unwrap().1)
-            .get_mut::<MouseButtons>()
-            .unwrap();
-        mouse_buttons.left = true;
+    pub fn on_left_click(&self, focused: bool) {
+        match focused {
+            true => {
+                let mut entities = self.entities.write();
+                let mut mouse_buttons = entities
+                    .world
+                    .entity_mut(self.player.read().unwrap().1)
+                    .get_mut::<MouseButtons>()
+                    .unwrap();
+                mouse_buttons.left = true;
+            }
+            false => self
+                .inventory_context
+                .write()
+                .on_click(self.renderer.clone()),
+        }
     }
 
-    pub fn on_release_left_click(&self) {
-        let mut entities = self.entities.write();
-        let mut mouse_buttons = entities
-            .world
-            .entity_mut(self.player.read().unwrap().1)
-            .get_mut::<MouseButtons>()
-            .unwrap();
-        mouse_buttons.left = false;
+    pub fn on_release_left_click(&self, focused: bool) {
+        if focused {
```

Comment:
this looks non-sensical

Reply:
The focused check, because why would you pass a bool to immediately start execution of the called function if it is false?

Reply:
I was trying to make each of these calls similar for the moment. Later on we can pass these event to the inventory window if focused is false.

Reply:
Okay, fair enough, a TODO would be nice tho

Reply:
I've added those comments

Reply:
I've added TODO comments to the end of those functions